src/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -13,15 +11,23 @@ namespace ts.server.typingsInstaller {
 
     class FileLog implements Log {
         private logEnabled = true;
-        constructor(private readonly logFile?: string) {
+        private readonly logFile: string;
+
+        constructor(logFile: string | undefined) {

```

Comment:
This constructor only assigns `this.logFile` in one of the two branches, so its type ought to be `string | undefined` and you will need `Debug.assertDefined(this.logFile)` in `writeLine`.

Reply:
- Updated `FileLog` class to always initialize `logFile` (the file name) property as a string.
- Check `logEnabled` (boolean) property before proceeding further in `writeLine()` method.

This should make the compiler fully happy!